src/main/java/hudson/plugins/sshslaves/SSHConnector.java

```diff
@@ -67,242 +66,187 @@
     /**
      * Field port
      */
-    public final int port;
+    public int port;
 
     /**
      * The id of the credentials to use.
      */
     private String credentialsId;
 
-    /**
-     * Transient stash of the credentials to use, mostly just for providing floating user object.
-     */
-    private transient StandardUsernameCredentials credentials;
-
-    /**
-     * Field username
-     *
-     * @deprecated
-     */
-    @Deprecated
-    public transient String username;
-
-    /**
-     * Field password
-     *
-     * @deprecated
-     */
-    @Deprecated
-    public transient Secret password;
-
-    /**
-     * File path of the private key.
-     *
-     * @deprecated
-     */
-    @Deprecated
-    public transient String privatekey;
-
     /**
      * Field jvmOptions.
      */
-    public final String jvmOptions;
```

Comment:
This fields wasn't deprecated and now they are removed

Reply:
encapsulation, those fields should be accessed by getter/setter methods to avoid mutation

Reply:
In jenkins world it's impossible just to remove something existing and public because it will break ALL things that already depending on this. That's why new is done right, old is untouched. Binary compatibility is (was?) one of the main jenkins features that developers/maintainers made a lot of efforts.